core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1928,21 +1928,17 @@ class Log(@volatile var dir: File,
 
   def nonActiveLogSegmentsFrom(from: Long): Iterable[LogSegment] = {
     lock synchronized {
-      if (from > activeSegment.baseOffset)
-        throw new IllegalArgumentException("Illegal request for non-active segments beginning at " +
-          s"offset $from, which is larger than the active segment's base offset ${activeSegment.baseOffset}")
-      logSegments(from, activeSegment.baseOffset)
+      if (from > activeSegment.baseOffset) {
```

Comment:
The log start offset may be ahead of the active segment base offset in the case of a call to DeleteRecords. It would be more appropriate/intuitive in this case to just return `Seq.empty`. Also we can get rid of the log warning.